lektor/quickstart.py

```diff
@@ -1,20 +1,25 @@
+import getpass
 import os
 import re
 import shutil
-import subprocess
-import tempfile
-import uuid
 from contextlib import contextmanager
 from datetime import datetime
 from functools import partial
+from importlib import import_module
+from subprocess import PIPE
+from subprocess import run
+from tempfile import TemporaryDirectory
+from typing import Optional
 
 import click
 from jinja2 import Environment
 from jinja2 import PackageLoader
 
-from lektor.utils import fs_enc
+from lektor.utils import locate_executable
 from lektor.utils import slugify
 
+pwd = import_module("pwd") if os.name != "nt" else None
```

Comment:
IMHO, the more common and simpler idiom to do this kind of conditional import is `try: ... except ImportError: ...`. Or does that not work in this case?

Reply:
I resorted to that syntax to make things clear for mypy.

Once mypy sees an `import` statement — whether in a conditional context or not —- it seems to treat that as _final_.

E.g.
```py
try:
    import pwd
except ImportError:
    pwd = None # Error: Incompatible types in assignment (expression has type "None", variable has type Module)
```

(The same error results if one does a conditional `if os.name != "nt":` around the import rather than the try...except.)

I also tried
```py
pwd: Optional[ModuleType] = None
if os.name != "nt":
    import pwd # Error: Name "pwd" already defined
```

----

Of note, because `sys.platform` is a mypy compile-time constant (as opposed to `os.name` which is not) this does work without error from mypy, since mypy will only "see" one leg or the other of the conditional:
```py
if sys.platform not in ("win32", "cygwin"):  # does cygwin have pwd?
    import pwd
else:
    pwd = None
```
However, I'm not sure that the `sys.platform` test is exactly what we want here.  (In any case Lektor seems to use `os.name` in favor of `sys.platform` for windows detection.) 

----

Anyhow, that's why I did that.  I'm open to alternatives.

Reply:
Thanks for the explanation and trying all these possible workarounds :)

Maybe this gets fixed in mypy at some point https://github.com/python/mypy/issues/1297 (there's a possible `try ... except ... else ...` workaround suggested in that thread but I don't think that that's much prettier)